src/chains/serializers.py

```diff
@@ -33,14 +33,25 @@ class CurrencySerializer(serializers.Serializer):
 
 
 class RpcUriSerializer(serializers.Serializer):
-    authentication = serializers.CharField(source="rpc_authentication")
-    value = serializers.URLField(source="rpc_uri")
+    def __init__(self, *args, **kwargs):
+        authentication_source = kwargs.pop(
+            "authentication_source", "rpc_authentication"
+        )
+        uri_source = kwargs.pop("uri_source", "rpc_uri")
+        self.fields.update(
+            {
+                "authentication": serializers.CharField(source=authentication_source),
+                "value": serializers.URLField(source=uri_source),
+            }
+        )
+        super(RpcUriSerializer, self).__init__(*args, **kwargs)
```

Comment:
I have a suggestion regarding this `__init__`. I think one advantage of declaring the fields in the serialiser is that it's easier to read what is actually returned on the json level. Therefore, what do you think of the following? `BaseRpcUriSerializer` is the base class which requires both `get_authentication` and `get_uri` to be implemented.

```python
class BaseRpcUriSerializer(serializers.Serializer):
    authentication = serializers.SerializerMethodField()
    uri = serializers.SerializerMethodField()

    @abstractmethod
    def get_authentication(self, obj):
        pass

    @abstractmethod
    def get_uri(self, obj):
        pass


class RpcUriSerializer(BaseRpcUriSerializer):

    def get_authentication(self, obj):
        return obj.rpc_authentication

    def get_uri(self, obj):
        return obj.rpc_uri


class SafeAppsRpcUriSerializer(BaseRpcUriSerializer):

    def get_authentication(self, obj):
        return obj.safe_apps_rpc_authentication

    def get_uri(self, obj):
        return obj.safe_apps_rpc_uri
```

Reply:
Te declaration on the root object would then have the following:

```python
    @staticmethod
    @swagger_serializer_method(serializer_or_field=BaseRpcUriSerializer)
    def get_safe_apps_rpc_uri(obj):
        return SafeAppsRpcUriSerializer(obj).data

    @staticmethod
    @swagger_serializer_method(serializer_or_field=BaseRpcUriSerializer)
    def get_rpc_uri(obj):
        return RpcUriSerializer(obj).data
```

Reply:
Thanks! Nice idea